minion/frontend/views.py

```diff
@@ -414,6 +422,72 @@ def persona_login():
     session['role'] = user['role']
     return api_session()
 
+def ldap_login(request):
+    data = json.loads(request.data)
+    username = data['user']
+    password = data['password']
+
+    # Initialize LDAP
+    try:
+        ld = ldap.initialize(config['login']['ldap']['uri'])
+    except ldap.LDAPError, e:
+        return jsonify(success=False, reason="error")
+
+    auth = config['login']['ldap']['uid_filter'] + "=%s," % username + config['login']['ldap']['base']
+
+    # Authentication
+    try:
+        ld.simple_bind_s(auth, password)
+    except ldap.LDAPError, e:
+        if e.message['desc'] == 'Invalid credentials':
+            return jsonify(success=False, reason="invalid_cred")
+        return jsonify(success=False, reason="error")
+
+    # Retrieve mail
+    searchScope = ldap.SCOPE_SUBTREE
+    results = ld.search_s(auth, searchScope, attrlist=[str(config['login']['ldap']['mail_filter'])])
+
+    try:
+        mail = results[0][1][config['login']['ldap']['mail_filter']][0]
+    except Exception as e:
+        return jsonify(sucess=False, reason="error")
+
+    if config['login']['ldap']['check_authorized_groups']:
+        # Check if user in an authorized group
+        group_base = config['login']['ldap']['group_base']
+
+        authorized = False
+        for group in config['login']['ldap']['authorized_groups']:
+            filter_group = "(&(objectClass=" + config['login']['ldap']['group_objectClass']+ ")"\
+					+ "(member=" + config['login']['ldap']['uid_filter'] + "=%s," %username \
+					+ config['login']['ldap']['base'] + ")"\
+					+ "(cn=%s))" % str(group)
+				
```

Comment:
This is not a very idiomatic way of doing this.  I would recommend something like:

    filter_group = ("(&"
                    "(objectClass={ldap[groupObjectClass]})"
                    "(member={ldap[usernameAttribute]})"
                    "(cn={group})"
                    ")".format(ldap=config['login']['ldap'], group=group))

This is a combination of string literal concatenation and using format() to its greatest effect.  It makes things a lot cleaner to read.  :)